zkevm-circuits/src/evm_circuit/execution/precompiles/ec_mul.rs

```diff
@@ -0,0 +1,256 @@
+            )?;
+            for (col, word_value) in [
+                (&self.point_p_x_rlc, aux_data.p_x),
+                (&self.point_p_y_rlc, aux_data.p_y),
+                (&self.scalar_s_rlc, aux_data.s),
+                (&self.point_r_x_rlc, aux_data.r_x),
+                (&self.point_r_y_rlc, aux_data.r_y),
+            ] {
+                col.assign(
+                    region,
+                    offset,
+                    keccak_rand.map(|r| rlc::value(&word_value.to_le_bytes(), r)),
+                )?;
+            }
+        }
+
+        self.is_success.assign(
+            region,
+            offset,
+            Value::known(F::from(u64::from(call.is_success))),
+        )?;
+        self.callee_address.assign(
+            region,
+            offset,
+            Value::known(call.code_address.unwrap().to_scalar().unwrap()),
+        )?;
+        self.caller_id
+            .assign(region, offset, Value::known(F::from(call.caller_id as u64)))?;
+        self.call_data_offset.assign(
+            region,
+            offset,
+            Value::known(F::from(call.call_data_offset)),
+        )?;
+        self.call_data_length.assign(
+            region,
+            offset,
+            Value::known(F::from(call.call_data_length)),
+        )?;
+        self.return_data_offset.assign(
+            region,
+            offset,
+            Value::known(F::from(call.return_data_offset)),
+        )?;
+        self.return_data_length.assign(
+            region,
+            offset,
+            Value::known(F::from(call.return_data_length)),
+        )?;
+
+        self.restore_context
+            .assign(region, offset, block, call, step, 7)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use bus_mapping::{
+        evm::{OpcodeId, PrecompileCallArgs},
+        precompile::PrecompileCalls,
+    };
+    use eth_types::{bytecode, ToWord};
+    use itertools::Itertools;
+    use mock::TestContext;
+    use rayon::iter::{ParallelBridge, ParallelIterator};
+
+    use crate::test_util::CircuitTestBuilder;
+
+    lazy_static::lazy_static! {
+        static ref TEST_VECTOR: Vec<PrecompileCallArgs> = {
+            vec![
+                PrecompileCallArgs {
+                    name: "ecMul (valid inputs)",
+                    // P = (1, 2)
+                    // s = 1
+                    setup_code: bytecode! {
+                        // p_x = 1
+                        PUSH1(0x01)
+                        PUSH1(0x00)
+                        MSTORE
+                        // p_y = 2
+                        PUSH1(0x02)
+                        PUSH1(0x20)
+                        MSTORE
+                        // s = 1
+                        PUSH1(0x01)
+                        PUSH1(0x40)
+                        MSTORE
+                    },
+                    call_data_offset: 0x00.into(),
+                    call_data_length: 0x60.into(),
+                    ret_offset: 0x80.into(),
```

Comment:
return data offset can be `0x60`, although these call args should be fine as well. Just a nit-pick

Reply:
You're right. Since the input is 96 bytes, the offset should be 0x60. This detail is corrected.